persistence/src/main/scala/hmda/persistence/processing/HmdaFileParser.scala

```diff
@@ -65,7 +71,23 @@ class HmdaFileParser(submissionId: String) extends PersistentActor with ActorLog
   override def receiveCommand: Receive = {
 
     case ReadHmdaRawFile(persistenceId) =>
-      val parsed = events(persistenceId)
+      var finished = false
+      val parsedTs = events(persistenceId)
+        .map { case LineAdded(_, data) => data }
+        .take(1)
+        .map(line => TsCsvParser(line))
+        .map {
+          case Left(errors) => TsParsedErrors(errors)
+          case Right(ts) => TsParsed(ts)
+        }
+
+      parsedTs
+        .runForeach(pTs => self ! pTs)
+        .andThen {
+          case _ => if (!finished) finished = true else self ! Shutdown
```

Comment:
Now that I'm looking at both parsing operations, I'm not sure I like this shutdown behavior here. It's probably easier to think about the message being responsible for reading the stream of data and sending parsing commands to itself, and leave it there. And then put the responsibility of shutting down on the caller. This would allow more than one read operation per actor instance. Probably requires some testing (i.e. manual testing through the API, to make sure behavior is consistent)